cheatsheets/Vulnerable_Dependency_Management_Cheat_Sheet.md

```diff
@@ -0,0 +1,233 @@
+
+Nothing specific because here we are in *patch yourself* condition.
+
+### Approach
+
+**Step 1:**
+
+If we are in this case due to one of the following condition, it's a good idea to start a parallel study to find another component better maintained or if it's a commercial component with support **then put pressure** on the provider with the help of your [Chief Risk Officer](https://en.wikipedia.org/wiki/Chief_risk_officer) (fallback possible to [Chief Information Security Officer](https://en.wikipedia.org/wiki/Chief_information_security_officer)):
+* Provider do not want to fix the issue.
+* Provider do not answer at all.
+
+In all cases, here, we need to handle the vulnerability right now...
+
+**Step 2:**
+
+As we know the vulnerable dependency, we know where it is used in the application (if it's a transitive dependency then we can identify the first level dependency using it using the [IDE](https://en.wikipedia.org/wiki/Integrated_development_environment) built-in feature or the dependency management system used (Maven, Gradle, Nuget, NPM...). Note that IDE is also used to identify call to the dependency. 
+
+So, we can identify any call to this dependency but we still miss information about which kind of patching we must perform...
+
+To obtain these informations, we use the CVE content to know which kind of vulnerability affect the dependency. The `description` property give the answer: SQL injection, Remote Code Execution, Cross-Site Scripting, Cross-Site Request Forgery...
+
+Now, we know which kind of patching we must perform (it's become similar to [Case 2](#case-2) with the protection code) and where to add it.
+
+*Example:*
+
+We have an application using the Jackson API in a version exposed to the [CVE-2016-3720](https://nvd.nist.gov/vuln/detail/CVE-2016-3720).
+
+The description of the CVE is:
+
+```text
+XML external entity (XXE) vulnerability in XmlMapper in the Data format extension for Jackson 
+(aka jackson-dataformat-xml) allows attackers to have unspecified impact via unknown vectors.  
+```
+
+Based on these information, we determine that the patching will be to add a [pre-validation of any XML data](XML_External_Entity_Prevention_Cheat_Sheet.md) passed to the Jakson API to prevent [XML external entity (XXE)](https://www.acunetix.com/blog/articles/xml-external-entity-xxe-vulnerabilities/) vulnerability.
+
+**Step 3:**
+
+If possible, create a unit test trying to trigger the vulnerability via your application in order to ensure that the patch is effective and have a way to continuously ensure that the patch is in place during the evolution of the project.
+
+If you have a set of automated unit or integration or functionnal or security tests that exists for the application then run them to verify that the patch do not impact the stability of the application.
+
+## Case 4
+
+### Context
+
+The vulnerable dependency in found via the discovery of a full disclosure post on Internet, provider not aware of the vulnerability.
```

Comment:
sometimes such bugs can be also found during penetration testing. For example I can use component 'X' that was newer pentested - I hire pentester to test my application - pentester can find bug in my application that is inside dependency 'X'

Reply:
Added to last case in coming commit.